docker-images/kafka/Dockerfile

```diff
@@ -56,6 +56,9 @@ COPY ./stunnel-scripts/ $STUNNEL_HOME
 
 # copy scripts for starting Kafka
 COPY ./scripts/ $KAFKA_HOME
+# download kafka-vault artifact and copy to its final destination
+RUN curl -L -O https://github.com/UltimateSoftware/kafka-vault-jca/releases/download/v1.0-develop/kafka-vault-jca-1.0-SNAPSHOT.jar \
```

Comment:
In the recent PR, we implemented loading of dependencies through Maven instead of just copying them. Can we get this class to Maven Central and download it through Maven? All what should be needed is to add it here: https://github.com/strimzi/strimzi-kafka-operator/blob/master/docker-images/kafka/kafka-thirdparty-libs/pom.xml